src/main/java/sirius/db/mixing/types/BaseEntityRef.java

```diff
@@ -180,6 +180,25 @@ public E fetchValue() {
         return value;
     }
 
+    /**
+     * Returns the effective entity object which is referenced using a secondary node of a DB cluster.
+     * <p>
+     * Note, this values returned by this method will never be cached and a new lookup is always performed.
+     *
+     * @return the entity being referenced or <tt>null</tt> if no entity is referenced.
+     */
+    public E fetchValueFromSecondary() {
+        if (value == null && id != null) {
+            Optional<E> entity = findInSecondary(type, id);
+            if (entity.isPresent()) {
+                return entity.get();
+            } else {
+                id = null;
+            }
```

Comment:
Would also not clear the ID in this case. It could be that a newly inserted entity just isn't propagated through the cluster yet.